diff --git a/cmd/age/age.go b/cmd/age/age.go
index f81d3bbad1..fdaed88173 100644
--- a/cmd/age/age.go
+++ b/cmd/age/age.go
@@ -471,6 +471,11 @@ func decrypt(identities []age.Identity, in io.Reader, out io.Writer) {
 }
 
 func passphrasePromptForDecryption() (string, error) {
+	if pass, exists := os.LookupEnv("AGE_PASSPHRASE"); exists {
+		fmt.Fprintln(os.Stderr, "Using passphrase from environment variable.")
+		return pass, nil
+	}
+
 	pass, err := readSecret("Enter passphrase:")
 	if err != nil {
 		return "", fmt.Errorf("could not read passphrase: %v", err)
